metabase/metabase
# Add global navigation sidebar

This PR is a foundation of a new global navigation layout for Metabase. It basically adds the sidebar we used to have on the collection page to every Metabase page, so it becomes accessible everywhere (with a few exceptions).

The implementation is a bit rough (it's mostly just `CollectionSidebar`) and the visual appearance is not yet final and will be changed in the subsequent PRs.

### How to review

The branch used to be a prototype branch and we didn't follow a feature branch approach, so the first half of commits is done directly on the branch without their own PRs. It'd make sense to review them commit by commit. The other half is done in a feature branch manner where every commit is a squashed PR that was reviewed separately.

### To Verify
_A great way to test this is to run a fresh Metabase installation to ensure the initial flow looks good too_

1. Go through as many Metabase pages as possible (query builder, dashboards, collections, browse data, admin, etc.) and for each page:
2. Make sure you can open/collapse the navigation sidebar (by clicking the chevron icon next to the Metabase logo at the top left) and make sure the layout doesn't look broken and every feature is available
3. Make sure the top bar and the sidebar are hidden in two cases:
    * when editing a dashboard
    * when editing a model's query/metadata

**Known issues**
1. Active element highlighting: the sidebar uses `CollectionSidebar` and it highlights selected collections by ID. Because of that, if you open "Browse data" or archive any bookmark, the sidebar won't reflect that. This is going to be fixed by #21298
2. BulkActionsBar layout (the one that appears when you select a few collection items for a bulk archive/move)
3. Mobile layout

### Demo

<img width="1472" alt="CleanShot 2022-03-29 at 14 41 14@2x" src="https://user-images.githubusercontent.com/17258145/160624801-4ca0033a-98e8-4c31-925e-c90cdd63a1b7.png">
